crates/zed/src/zed.rs

```diff
@@ -634,14 +634,19 @@ fn reload_keymaps(cx: &mut AppContext, keymap_content: &KeymapFile) {
 }
 
 pub fn load_default_keymap(cx: &mut AppContext) {
-    KeymapFile::load_asset(DEFAULT_KEYMAP_PATH, cx).unwrap();
-    if VimModeSetting::get_global(cx).0 {
-        KeymapFile::load_asset("keymaps/vim.json", cx).unwrap();
-    }
-
-    if let Some(asset_path) = BaseKeymap::get_global(cx).asset_path() {
+    let base_keymap = *BaseKeymap::get_global(cx);
```

Comment:
This logic seems a bit convoluted (you're relying on the fact that asset_path is only null for BaseKeymap::None). I'd prefer to keep the special-case more obvious and retain the previous structure:

```
let base_keymap = *BaseKeymap::get_global(cx);
if base_keymap == BaseKeymap::None {
  return
}

KeymapFile::load_asset(DEFAULT_KEYMAP_PATH, cx).unwrap();
if VimModeSetting::get_global(cx).0 {
    KeymapFile::load_asset("keymaps/vim.json", cx).unwrap();
}
if let Some(asset_path) = base_keymap.asset_path() {
   KeymapFile::load_asset(asset_path, cx).unwrap();
}
```

Reply:
fair point!